### PR TITLE
76 pub sub

### PR DESCRIPTION
**Pull request by if0s**

### CODE REVIEW

File: component.json [hunk 1]
```diff
@@ -253,6 +253,52 @@
           "prompt": "Please select a Event object"
         }
       }
+    },
+    "streamPlatformEventsPubSub": {
+      "title": "Subscribe to PubSub",
+      "order": 97,
+      "help": {
+        "description": "Can be used for subscription to the specified in the configuration Platform Event object. Can be used only for Realtime flows",
```
Comment (A3a3e1) [inline component.json:261]:
```suggestion
        "description": "Can be used to subscribe to an object specified in the configuration Platform Event. **Can be used in Realtime flows only**",
```

File: component.json [hunk 2]
```diff
@@ -253,6 +253,52 @@
...
+      },
+      "main": "./lib/triggers/streamPlatformEventsPubSub.js",
+      "type": "polling",
+      "fields": {
+        "object": {
+          "viewClass": "SelectView",
+          "label": "Event object name",
+          "required": true,
+          "model": "objectTypes",
+          "prompt": "Please select a Event object"
```
Comment (A3a3e1) [inline component.json:272]:
```suggestion
          "prompt": "Please select Event object"
```

File: component.json [hunk 3]
```diff
@@ -253,6 +253,52 @@
...
+          "label": "Pub/Sub API Endpoint",
+          "viewClass": "TextFieldView",
+          "required": false,
+          "placeholder": "api.pubsub.salesforce.com:7443",
+          "help": {
+            "description": "You can set Pub/Sub API Endpoint manually or leave it blank for default: \"api.pubsub.salesforce.com:7443\""
+          }
+        },
+        "eventCountPerRequest": {
+          "label": "Number of events per request",
```
Comment (A3a3e1) [inline component.json:284]: Are there any upper limits?

File: component.json [hunk 4]
```diff
@@ -253,6 +253,52 @@
...
+            "description": "10"
+          }
+        },
+        "initialReplayId": {
+          "label": "Start from Replay Id",
+          "viewClass": "TextFieldView",
+          "required": false,
+          "placeholder": "31142963",
+          "help": {
+            "description": "1"
```
Comment (A3a3e1) [inline component.json:298]: This is an interesting description :)

File: component.json (hunk 3 above)
Comment (A3a3e1) [reply component.json:284]: OK. Found this in the code 'Number of events per request" must be valid number between 1 and 100'. It makes sense to write it somewhere here

File: lib/triggers/streamPlatformEventsPubSub.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  if (!secretId) {
+    context.logger.error('secretId is missing in configuration, credentials cannot be fetched');
+    creationInProgress = false;
+    throw new Error('secretId is missing in configuration, credentials cannot be fetched');
+  }
+  if (refreshTokenNeeded) {
+    await refreshToken(context, secretId, msg.id);
+    refreshTokenNeeded = false;
+  }
+  context.logger.info('Starting "Subscribe to PubSub" trigger');
```
Comment (A3a3e1) [inline lib/triggers/streamPlatformEventsPubSub.js:75]: This should come at the very beginning of the trigger

File: component.json (hunk 1 above)
Comment (if0s) [reply component.json:261]: `Can be used to subscribe to an object specified in the configuration Platform Event.` - changed
`Can be used in Realtime flows only!` - removed (this one can work as ordinary)

File: component.json (hunk 2 above)
Comment (if0s) [reply component.json:272]: done

File: component.json (hunk 3 above)
Comment (if0s) [reply component.json:284]: done

File: component.json (hunk 4 above)
Comment (if0s) [reply component.json:298]: fixed

File: lib/triggers/streamPlatformEventsPubSub.js (hunk 5 above)
Comment (if0s) [reply lib/triggers/streamPlatformEventsPubSub.js:75]: moved

**Review by A3a3e1 — APPROVED** [review]

**Review by HannaTrotsenko — APPROVED** [review]

**Pull request merged**